data/settings.js

```diff
@@ -515,123 +515,290 @@ depending on the authenticated user).`
-
-See [[setting,sieve_after]] for configuration details, as this
-setting behaves the same way, except the scripts are run **before** user's
-personal scripts (instead of **after**).`
+A file system path pointing to a Sieve script file or a directory containing one
+or more Sieve script files with names structured as \`<script-name>.sieve\`.
+This setting only applies when \`sieve_script_driver = file\``
 	},
 
-	sieve_default: {
-		tags: [ 'sieve' ],
+	sieve_script_active_path: {
+		tags: [ 'sieve-storage-file' ],
 		plugin: 'sieve',
+		default: '~/.dovecot.sieve',
+		seealso: [ '[[link,sieve_storage_file]]' ],
 		values: setting_types.STRING,
 		text: `
-[[link,sieve_location,Location]] of the default personal sieve script file
-which gets executed ONLY if user's private Sieve script does not exist, e.g.
-\`file:/var/lib/dovecot/default.sieve\` (check the
-[[link,sieve_multiscript,multiscript section]]
-for instructions on running global Sieve scripts before and after the user's
-personal script).
+When [[link,managesieve_server]] is used, one script in the storage can
+be active; i.e., evaluated at delivery.
+
+This setting only applies when \`sieve_script_driver = file\`. For that storage
+driver, the active script in the storage directory is pointed to by a symbolic
+link.
 
-This is usually a global script, so be sure to pre-compile the specified
-script manually in that case using the sievec command line tool, as
-explained by [[man,sievec]].`
+This setting configures where this symbolic link is located. If the
+\`sieve_script_path\` setting points to a regular file, this setting has no
+effect (and ManageSieve cannot be used).`
 	},
 
-	sieve_default_name: {
-		tags: [ 'sieve' ],
+	sieve_script_ldap_script_attr: {
+		tags: [ 'sieve-storage-ldap' ],
 		plugin: 'sieve',
+		default: 'mailSieveRuleSource',
+		seealso: [ '[[link,sieve_storage_ldap]]' ],
 		values: setting_types.STRING,
-		seealso: [ '[[link,sieve_visible_default_script]]' ],
 		text: `
-The name by which the default Sieve script is visible to ManageSieve
-clients. Normally, it is not visible at all.`
+The name of the attribute containing the Sieve script.
+`
 	},
 
-	sieve_discard: {
-		tags: [ 'sieve' ],
+	sieve_script_ldap_mod_attr: {
+		tags: [ 'sieve-storage-ldap' ],
 		plugin: 'sieve',
+		default: 'modifyTimestamp',
+		seealso: [ '[[link,sieve_storage_ldap]]' ],
 		values: setting_types.STRING,
 		text: `
-The location of a Sieve script that is run for any message that is about
-to be discarded; i.e., it is not delivered anywhere by the normal Sieve
-execution.
+The name of the attribute used to detect modifications to the LDAP entry.`
+	},
 
-This only happens when the "implicit keep" is canceled, by e.g. the
-"discard" action, and no actions that deliver the message are executed.
+	sieve_script_ldap_filter: {
+		tags: [ 'sieve-storage-ldap' ],
+		plugin: 'sieve',
+		default: '(&(objectClass=posixAccount)(uid=%{user}))',
```

Comment:
We removed such default ldap filters from auth-ldap. It's unlikely anyone will use this default filter with LDAP. So lets remove such defaults also from sieve-ldap?

Reply:
Should all be changed in separate LDAP ticket.

Reply:
done